Testing/VelaptorTests/Services/BatchServiceManagerTests.cs

```diff
@@ -0,0 +1,504 @@
+        {
+            // Arrange
+            var batchItem = default(FontGlyphBatchItem);
+            this.mockFontGlyphBatchingService.Setup(m => m.Add(It.IsAny<FontGlyphBatchItem>()))
+                .Callback<FontGlyphBatchItem>((_) =>
+                {
+                    this.mockFontGlyphBatchingService.Raise(e => e.BatchFilled += null, EventArgs.Empty);
+                });
+
+            var manager = CreateManager();
+
+            // Act & Assert
+            Assert.Raises<EventArgs>(e =>
+            {
+                manager.FontGlyphBatchFilled += e;
+            }, e =>
+            {
+                manager.FontGlyphBatchFilled -= e;
+            }, () =>
+            {
+                manager.AddFontGlyphBatchItem(batchItem);
+            });
+
+            this.mockFontGlyphBatchingService.VerifyOnce(m => m.Add(batchItem));
+        }
+
+        [Fact]
+        public void AddRectBatchItem_WithFullBatch_RaisesFontGlyphBatchFilledEvent()
+        {
+            // Arrange
+            var batchItem = default(RectShape);
+            this.mockRectBatchingService.Setup(m => m.Add(It.IsAny<RectShape>()))
+                .Callback<RectShape>((_) =>
+                {
+                    this.mockRectBatchingService.Raise(e => e.BatchFilled += null, EventArgs.Empty);
+                });
+
+            var manager = CreateManager();
+
+            // Act & Assert
+            Assert.Raises<EventArgs>(e =>
+            {
+                manager.RectBatchFilled += e;
+            }, e =>
+            {
+                manager.RectBatchFilled -= e;
+            }, () =>
+            {
+                manager.AddRectBatchItem(batchItem);
+            });
+
+            this.mockRectBatchingService.VerifyOnce(m => m.Add(batchItem));
+        }
+
+        [Fact]
+        public void EndBatch_WhenTextureBatch_RaisesTextureBatchFilledEvent()
+        {
+            // Arrange
+            var manager = CreateManager();
+
+            // Act & Assert
+            Assert.Raises<EventArgs>(e =>
+            {
+                manager.TextureBatchFilled += e;
+            }, e =>
+            {
+                manager.TextureBatchFilled -= e;
+            }, () =>
+            {
+                manager.EndBatch(BatchServiceType.Texture);
+            });
+        }
+
+        [Fact]
+        public void EndBatch_WhenFontGlyphBatch_RaisesFontGlyphBatchFilledEvent()
+        {
+            // Arrange
+            var manager = CreateManager();
+
+            // Act & Assert
+            Assert.Raises<EventArgs>(e =>
+            {
+                manager.FontGlyphBatchFilled += e;
+            }, e =>
+            {
+                manager.FontGlyphBatchFilled -= e;
+            }, () =>
+            {
+                manager.EndBatch(BatchServiceType.FontGlyph);
+            });
+        }
+
+        [Fact]
+        public void EndBatch_WhenRectBatch_RaisesFontGlyphBatchFilledEvent()
```

Comment:
Refactor unit test name.  Name incorrect.

```suggestion
        public void EndBatch_WhenRectBatch_RaisesRectBatchFilledEvent()
```